src/app/Http/Controllers/Auth/VerifyEmailController.php

```diff
@@ -60,10 +69,28 @@ public function verifyEmail(EmailVerificationRequest $request)
      */
     public function resendVerificationEmail(Request $request): \Illuminate\Http\RedirectResponse
     {
-        $request->user(backpack_guard_name())->sendEmailVerificationNotification();
+        $user = $this->getUser($request);
 
+        if(! $user) {
+            return redirect()->route('backpack.auth.login');
+        }
+            
+        $user->sendEmailVerificationNotification();
         Alert::success('Email verification link sent successfully.')->flash();
 
         return back()->with('status', 'verification-link-sent');
     }
+
+    private function getUser(Request $request): ?\Illuminate\Contracts\Auth\MustVerifyEmail
+    {
+        return $request->user(backpack_guard_name()) ?? $this->getUserFromCookie();
+    }
+    
+    private function getUserFromCookie(): ?\Illuminate\Contracts\Auth\MustVerifyEmail
+    {
+        if (Cookie::has('backpack_email_verification')) {
+            return config('backpack.base.user_model_fqn')::where('email', Cookie::get('backpack_email_verification'))->first();
```

Comment:
Shouldn't we also use `config('backpack.base.email_column')` here?

```suggestion
            return config('backpack.base.user_model_fqn')::where(config('backpack.base.email_column'), Cookie::get('backpack_email_verification'))->first();
```